ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestination.java

```diff
@@ -162,7 +167,8 @@ public boolean updateBlobExpiration(BlobId blobId, long expirationTime) throws C
     }
 
     // For single blob GET request, get metadata from ABS instead of Cosmos
-    if (blobIds.size() == 1) {
+    // Note: findMissingKeys needs to query Cosmos regardless
```

Comment:
Could you expand on why the cosmos query is needed for findMissingKeys in this comment?